larray_editor/editor.py

```diff
@@ -396,11 +396,14 @@ def void_formatter(array, *args, **kwargs):
                 self.new()
         # convert input data to Session if not
         else:
-            self.data = data if isinstance(data, Session) else Session(data)
-            if qtconsole_available:
-                self.kernel.shell.push(dict(self.data.items()))
-            arrays = [k for k, v in self.data.items() if self._display_in_grid(k, v)]
-            self.add_list_items(arrays)
+            self._push_data(data)
```

Comment:
Is it intended that "self._listwidget.setCurrentRow(0)" is not longer called in the self.new() case above? It is probably fine but have you checked that?

Reply:
I don't see the reason to call "self._listwidget.setCurrentRow(0)" in self.new() ?
I checked and no exception was raised.